sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2565,7 +2580,7 @@ class AstBuilder extends DataTypeAstBuilder
     } else {
       // It's a function call
       val funcCtx = ctx.functionName
-      val func = withFuncIdentClause(
+      val func: Expression = withFuncIdentClause(
```

Comment:
unnecessary change?

Reply:
Nope, there are two withFuncIdentClause, one that takes LogicalPlan and other with Expression. Compiler was not smart to pick up which one it was.